src/hooks/useTodoList.ts

```diff
@@ -14,10 +16,19 @@ export const useTodoList = () => {
    *  - Saves the new todo in the state.
    * @param {TodoType} todo todo which should be added
    */
-  const addNewTodo = useCallback((todo: TodoType): void => {
-    setMyTodos((prev) => {
-      return [...prev, todo];
-    });
+
+  const addNewTodo = useCallback(async (todoItem: TodoType) => {
```

Comment:
```suggestion
  const addNewTodo = useCallback(async (title: string) => {
```
Implement this change